drivers/include/net/netdev.h

```diff
@@ -290,8 +305,32 @@ struct netdev {
 #ifdef MODULE_L2FILTER
     l2filter_t filter[CONFIG_L2FILTER_LISTSIZE];   /**< link layer address filters */
 #endif
+#if !IS_ACTIVE(GNRC_NETIF_SINGLE)
```

Comment:
I don't like this reverse dependency for GNRC in `netdev` here. How about introducing a pseudo-module for these members and the `netdev_register()` function, that is included when `GNRC_NETIF_SINGLE` is not defined?

Reply:
So replace `GNRC_NETIF_SINGLE` with a `netif_single` pseudo-module?
Sounds good, I'll make a separate PR.

Reply:
No, I meant a `netdev_register` pseudo-module.

Reply:
`netif_single` requires to much rework in the lower layers ;-).

Reply:
Then I don't understand.
How would a `netdev_register` pseudo-module help?
The idea was that we don't need to keep track of netif IDs if there is only a single netif.
But that information is currently only provided by `GNRC_NETIF_SINGLE`

Reply:
> How would a `netdev_register` pseudo-module help?

```suggestion
#if IS_USED(MODULE_NETDEV_REGISTER)
```

And have `netdev_register` be

```C
static inline void netdev_register(struct netdev *dev, netdev_type_t type, uint8_t index)
{
#if IS_USED(MODULE_NETDEV_REGISTER)
    dev->type  = type;
    dev->index = index;
#else
    (void) dev;
    (void) type;
    (void) index;
#endif
}
```

This way you can still optimize it out, but you don't have the implicit dependency to GNRC here.

Reply:
So

```Make
ifeq (,$(filter gnrc_netif_single,$(USEMODULE)))
  USEMODULE += netdev_register
endif
```

Reply:
Yepp.